4.14 GA Bugs for Baremetal, Etcd, and Management Console (#66216)
openshift/openshift-docs: release_notes/ocp-4-14-release-notes.adoc

[id="ocp-4-14-bare-metal-hardware-bug-fixes"]
==== Bare Metal Hardware Provisioning

> * Previously, if the hostname of a bare-metal machine was not provided by either reverse DNS or DHCP, it would default to `localhost` during bare-metal cluster provisioning on installer-provisioned infrastructure. This issue caused Kubernetes node name conflicts and prevented the cluster from being deployed. Now, if the hostname is detected to be `localhost` the provisioning agent sets the persistent hostname to the name of the `BareMetalHost` object. (link:https://issues.redhat.com/browse/OCPBUGS-9072[*OCPBUGS-9072*])

Reviewer: ```suggestion
* Previously, if the hostname of a bare-metal machine was not provided by either reverse DNS or DHCP, it would default to `localhost` during bare-metal cluster provisioning on installer-provisioned infrastructure. This issue caused Kubernetes node name conflicts and prevented the cluster from being deployed. Now, if the hostname is detected to be `localhost`, the provisioning agent sets the persistent hostname to the name of the `BareMetalHost` object. (link:https://issues.redhat.com/browse/OCPBUGS-9072[*OCPBUGS-9072*])
```
Ending comma missing from clause

[id="ocp-4-14-management-console-bug-fixes"]
==== Management Console

> * Previously, alerts loaded from non-prometheus datasources such as logs. This caused the source of all alerts to be displayed always as *Prometheus*. With this update, alert sources are displayed correctly. (link:https://issues.redhat.com/browse/OCPBUGS-9907[*OCPBUGS-9907*])

Reviewer: ```suggestion
* Previously, alerts loaded from non-Prometheus datasources such as logs. This caused the source of all alerts to be displayed always as *Prometheus*. With this update, alert sources are displayed correctly. (link:https://issues.redhat.com/browse/OCPBUGS-9907[*OCPBUGS-9907*])
```
Looks like a brand name that needs a cap. https://docs.google.com/spreadsheets/d/1DLS_lS3VKidgZIvcLmLp9BoiqptkvqHWfe1D5FD2kfk/edit#gid=1375785039

> * Previously, there was an issue with Patternfly 4 where you could not select or change the log component under the master node's logs section once a selection was already made. With this update, when you change to the log component from master node's log section, refresh the page to reload the default options. (link:https://issues.redhat.com/browse/OCPBUGS-18727[*OCPBUGS-18727*])

Reviewer: ```suggestion
* Previously, there was an issue with Patternfly 4 where you could not select or change the log component under the logs section of master node once a selection was already made. With this update, when you change to the log component from the log section of the master node, refresh the page to reload the default options. (link:https://issues.redhat.com/browse/OCPBUGS-18727[*OCPBUGS-18727*])
```

* Previously, an empty page was displayed when viewing route details on the *Metrics* tab of the *`alertmanager-main`* page. With this update, user privileges were updated so you can view the route details on the *Metrics* tab. (link:https://issues.redhat.com/browse/OCPBUGS-15021[*OCPBUGS-15021*])

> * Previously, {product-rosa} used custom branding and the favicon would disappear so no specific branding appeared when custom branding is in use. With this update, {product-rosa} branding is now part of the branding API. (link:https://issues.redhat.com/browse/OCPBUGS-14716[*OCPBUGS#14716*])

Reviewer: ```suggestion
* Previously, {product-rosa} used custom branding and the favicon would disappear so no specific branding appeared when custom branding was being used. With this update, {product-rosa} branding is now part of the branding API. (link:https://issues.redhat.com/browse/OCPBUGS-14716[*OCPBUGS#14716*])
```

> * Previously, the {product-title} web console did not render the monitoring *Dashboard* page when a Proxy was expected, and as a result websocket connection failed. With this update, the web console also detects proxy settings from environment variables. (link:https://issues.redhat.com/browse/OCPBUGS-14550[*OCPBUGS-14550*])

Reviewer: ```suggestion
* Previously, the {product-title} web console did not render the monitoring *Dashboard* page when a proxy was expected. As a result, the websocket connection failed. With this update, the web console also detects proxy settings from environment variables. (link:https://issues.redhat.com/browse/OCPBUGS-14550[*OCPBUGS-14550*])
```

* Previously, if the `console.openshift.io/disable-operand delete: "true"` and `operator.openshift.io/uninstall-message: "some message"` annotations were used on an operator CSV, the uninstall instructions did not show up in the web console. With this update, the instructions to opt out of the installment are available. (link:https://issues.redhat.com/browse/OCPBUGS-13782[*OCPBUGS-13782*])

> * Previously, the size on the *PersistentVolumeClaims* namespace *Details* page was incorrect. With this update, the prometheus query on *PersistentVolumeClaims* namespace *Details* page includes the namespace label and the size is now correct. (link:https://issues.redhat.com//browse/OCPBUGS-13208[*OCPBUGS-13208*])

Reviewer: ```suggestion
* Previously, the size on the *PersistentVolumeClaims* namespace *Details* page was incorrect. With this update, the Prometheus query on *PersistentVolumeClaims* namespace *Details* page includes the namespace label and the size is now correct. (link:https://issues.redhat.com//browse/OCPBUGS-13208[*OCPBUGS-13208*])
```

* Previously, after customizing the routes for console and downloads, the downloads route did not update in the `ConsoleCLIDownloads` link and pointed to the default downloads route. With this update, the `ConsoleCLIDownloads` link updates when the custom downloads route is set. (link:https://issues.redhat.com/browse/OCPBUGS-12990[*OCPBUGS-12990*])

* Previously, the print preview displayed incomplete topology information from the list view. With this update, a full list of resources is printed when they are longer than one page. (link:https://issues.redhat.com/browse/OCPBUGS-11219[*OCPBUGS-11219*])

> * Previously, dynamic plugins that proxy to services with longer response times timed out at 30 seconds with a 504 response. With this updatee, a 5 minute haproxy timeout annotation was added to the console route to match most browsers maximum timeout. (link:https://issues.redhat.com/browse/OCPBUGS-9917[*OCPBUGS-9917*])

Reviewer: ```suggestion
* Previously, dynamic plugins that proxy to services with longer response times timed out at 30 seconds with a 504 error message. With this update, a 5-minute HAProxy timeout annotation was added to the console route to match the maximum timeout of most browsers. (link:https://issues.redhat.com/browse/OCPBUGS-9917[*OCPBUGS-9917*])
```
https://docs.openshift.com/container-platform/3.6/architecture/networking/haproxy-router.html for style

> * Previously, the provided API page was using the `displayName` of the provided API which was not always set. As a result, the list was empty but you could still click all instances to get to the new instances YAML. With this update, the provided API if the displayName is not set and the list now displays text. (link:https://issues.redhat.com/browse/OCPBUGS-8682[*OCPBUGS-8682*])

Reviewer: ```suggestion
* Previously, the provided API page used the `displayName` of the provided API, but this value was not always set. As a result, the list was empty but you could still click all instances to get to the YAML of a new instance. With this update, if the `displayName` is not set, the list displays text. (link:https://issues.redhat.com/browse/OCPBUGS-8682[*OCPBUGS-8682*])
```

> * Previously, the `CronJobs` table and details view did not have a suspend indication. With this update, `spec.suspend` was added to the list and details view for `CronJobs`. (link:https://issues.redhat.com/browse/OCPBUGS-8299[*OCPBUGS-8299*])

Reviewer: ```suggestion
* Previously, the `CronJobs` table and details view did not have a `suspend` indication. With this update, `spec.suspend` was added to the list and details view for `CronJobs`. (link:https://issues.redhat.com/browse/OCPBUGS-8299[*OCPBUGS-8299*])
```
Based on the bug text, it looks like suspend needs back ticks.

> * Previously, when enabling a single plugin in the console operator's configuration, the redeployed console will fail. With this update, the list of plugins is now unique and pods run as expected. (link:https://issues.redhat.com/browse/OCPBUGS-5059[*OCPBUGS-5059*])

Reviewer: ```suggestion
* Previously, when enabling a single plugin in the configuration of the console operator, the redeployed console fails. With this update, the list of plugins is now unique and pods run as expected. (link:https://issues.redhat.com/browse/OCPBUGS-5059[*OCPBUGS-5059*])
```
Avoid possessives where possible and future tense.
